examples/mqtt/README.md

```diff
@@ -0,0 +1,80 @@
+# Using MQTT as Transport in LIOTA
+
+LIOTA offers MQTT protocol as transport at both Device & DCC ends via ![MqttDeviceComms](/liota/device_comms/mqtt_device_comms.py) & ![MqttDccComms](/liota/dcc_comms/mqtt_dcc_comms.py).
+
+
+## QoS
+LIOTA also supports configuration of QoS related parameters like in_flight size, queue_size and retry timeout using **QoSDetails class**.
+
+
+## Credentials and TLS Configurations
+* **Credentials class** encapsulates certificates or credentials related to a connection used at both DCC and Device side.
+* **TLSConf class** encapsulates parameters related to TLS configuration.
+
+![Example](/examples/mqtt/dcc_comms/aws_iot/simulated_home_topic_per_metric.py) shows how the above objects can be used to establish a secure connection.
+
+
+## Using MqttDeviceComms
+
+For both ProjectICE & non-ProjectICE use-cases, MQTT related parameters required in `publish()` and `subscribe()` like topic, qos, etc., are fetched directly from the user's
+configuration file.  Please refer this ![example](/examples/mqtt/iot_cc/iotcc_simulated_mqtt.py) for detailed explanation.
+
+
+## Using MqttDccComms
+
+MqttDccComms provides the flexibility of having a single topic (auto-generated or from configuration file) per MQTT connection or individual topic per metric. **MqttMessagingAttributes class** in ![mqtt.py](/liota/lib/transports/mqtt.py)
+provides this flexibility.
+
+### MqttMessagingAttributes
+
+It enables the following options for developers in LIOTA.
+
+**(a)** Use single publish and subscribe topic generated by LIOTA for an EdgeSystem, its Devices and its Metrics. By default:
+* Publish topic for all Metrics will be **liota/generated_local_uuid_of_edge_system**
```

Comment:
I think we can refactor it more:
liota/gw-uuid/requests
& 
liota/gw-uuid/response

Gives much more subscription possibilities.